src/traces/bar/plot.js

```diff
@@ -534,6 +537,64 @@ function getTextPosition(trace, index) {
     return helpers.coerceEnumerated(attributeTextPosition, value);
 }
 
+function calcTexttemplate(fullLayout, calcTrace, index, xa, ya) {
```

Comment:
It would be nice to merge this routine with `calcTextinfo` where we could first set the `obj.??Label` values and then use those to translate the `textinfo` flags.